### PR TITLE
Handle user search not found error

### PR DESCRIPTION
**Pull request by tfussell**

### CODE REVIEW

File: src/component/app/index.tsx [hunk 1]
```diff
@@ -1,37 +1,36 @@
+import { useAuth0 } from "@auth0/auth0-react";
```
Comment (tfussell) [inline src/component/app/index.tsx:1]: Using `JS/TS Import/Export Sorter` VS Code Extension to keep these alphabetically ordered and grouped by external then local imports (like `goimports`).

File: src/component/app/index.tsx [hunk 2]
```diff
@@ -70,29 +69,22 @@ export function Component() {
...
+  if (authLoading) {
     return <span>Checking auth...</span>;
-  }
-
-  if (error) return <Navigate to={`signin`} />;
-
-  if (!isLoading && !isAuthenticated) return <Navigate to={`signin`} />;
-
-  if (userLoading || isLoading || userStatus === "idle")
+  } else if (authError || (!authLoading && !isAuthenticated)) {
```
Comment (tfussell) [inline src/component/app/index.tsx:83]: Cleaned up the boolean logic here. Should be equivalent.

File: src/component/app/index.tsx [hunk 3]
```diff
@@ -104,7 +96,7 @@ function AuthenticatedRoute() {
   return (
     <UserContext.Provider value={userContext}>
       <Routes>
-        <Route path="profile" element={<Profile userLoading={userLoading} />} />
+        <Route path="profile" element={<Profile userLoading={false} />} />
```
Comment (tfussell) [inline src/component/app/index.tsx:99]: Must be `false` due to the above checks. Is this intended?

File: src/module/hook/user/index.ts [hunk 4]
```diff
@@ -33,8 +34,16 @@ const getAllUser = async (searchAllUser: ISearchAllUser) => {
...
+  try {
+    const data = await api.API.User.Search(searchCurrentUser);
+    return data;
+  } catch (err) {
+    if (err.message === 'index must not be empty') {
+      // indicates that the user doesn't exist in the DB yet, not an error
+      return [];
+    }
+    throw err;
+  }
```
Comment (tfussell) [inline src/module/hook/user/index.ts:46]: Main change here.

File: src/component/app/index.tsx (hunk 3 above)
Comment (marcusellison) [reply src/component/app/index.tsx:99]: nope, not intended.

Comment (tfussell) [reply src/component/app/index.tsx:103]: We have
```js
  } else if (userLoading || userStatus === "idle") {
    return <span>User loading</span>;
  }
```
above. Should we drop that to let the `Profile` component render whether the user is loading or not?

**Review by marcusellison — APPROVED** [review]

**Pull request merged**